src/random_test.cpp

```diff
@@ -51,11 +51,27 @@ static std::string get_result(std::string sql, Thd1 *thd) {
   return result;
 }
 
-/* return the db branch */
-static std::string db_branch() {
-  std::string branch = mysql_get_client_info();
-  branch = branch.substr(0, 3); // 8.0 or 5.7
-  return branch;
+/* return PS version in number format
+ Example 8.0.26 -> 80026
+ Example 5.7.35 -> 50735
+*/
+static unsigned long ps_version() {
+  std::string ps_base = mysql_get_client_info();
+  unsigned long major = 0, minor = 0, version = 0;
+  std::size_t major_p = ps_base.find(".");
+  if(major_p != std::string::npos) major = stoi(ps_base.substr(0, major_p));
+
+  std::size_t minor_p = ps_base.find(".", major_p + 1);
+  if (minor_p != std::string::npos)
+    minor = stoi(ps_base.substr(major_p + 1, minor_p - major_p));
+
+  std::size_t version_p = ps_base.find(".", minor_p + 1);
+  if (version_p != std::string::npos)
+    version = stoi(ps_base.substr(minor_p + 1, version_p - minor_p));
```

Comment:
check if it should be version or version_p

Reply:
It should be "version". The existing code is fine.